src/store/Chat/index.js

```diff
@@ -161,9 +180,34 @@ function ChatReducer (state = initialState, action) {
       //   });
       // }
 
+      // We need to run through the entire array of messages and aggregate them
+      //  into similar sets
+
+      if (action.payload.length > 0) {
+        sortedPayload = action.payload.sort((curr, next) => (
+          new Date(curr.timestamp) - new Date(next.timestamp)
+        ));
+        // There should be an algorithm here that would speed things up
+
+        for (let i = 0; i < sortedPayload.length; i += 1) {
+          // All we have to do is see if the last message has the same author
```

Comment:
Haha after writing it, its actually _super_ simple! It just looks hard to digest